modules/openy_gc_log/src/LogArchiver.php

```diff
@@ -143,6 +145,18 @@ protected function loadLogEntities() {
     return $this;
   }
 
+  /**
+   * Prepare directory for logs.
+   */
+  protected function prepareFileDirectory($dir) {
+    $dir = self::BASE_ARCHIVE_PATH . DIRECTORY_SEPARATOR . $dir;
+    if (!file_exists($dir)) {
```

Comment:
Let's use drupal utilities for this to simplify code:
- [FileSystem::mkdir](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21File%21FileSystem.php/function/FileSystem%3A%3Amkdir/8.2.x)
- [file_prepare_directory](https://api.drupal.org/api/drupal/core%21includes%21file.inc/function/file_prepare_directory/8.2.x) - please check, maybe it deprecated